instance/tests/models/factories/openedx_appserver.py

```diff
@@ -37,4 +37,10 @@ def make_test_appserver(instance=None):
     if not instance.load_balancing_server:
         instance.load_balancing_server = LoadBalancingServer.objects.select_random()
         instance.save()
-    return instance._create_owned_appserver()
+    appserver = instance._create_owned_appserver()
+
+    if server:
```

Comment:
This doesn't fix the problem of repeat code I mentioned -- we're still just repeating the creation of `ReadyOpenStackServerFactory` everywhere.

What we should do instead is, `appserver.server = ReadyOpenStackServerFactory() if server is None else server`.

Reply:
Well, I changed the default to Ready state, and 20 tests started failing, because the default server being created is in the Pending state, and that's why we have to overwrite it with the Ready state server. I think it's suitable to use it they what I have it now because it's very similar to how `instance` is being used.

Reply:
Ah, OK, I did not know about that default. Sorry, my recommendation was not very well informed here. But the addition to the make_test_appserver function is still pretty nice, thanks!